core/src/test/java/io/grpc/RoundRobinLoadBalancerTest.java

```diff
@@ -0,0 +1,154 @@
+  private LoadBalancer<Transport> loadBalancer;
+
+  private List<List<ResolvedServerInfo>> servers;
+  private List<EquivalentAddressGroup> addressGroupList;
+
+  @Mock private TransportManager<Transport> mockTransportManager;
+  @Mock private Transport mockTransport0;
+  @Mock private Transport mockTransport1;
+  @Mock private Transport mockTransport2;
+  @Mock private InterimTransport<Transport> mockInterimTransport;
+  @Mock private Transport mockInterimTransportAsTransport;
+  @Captor private ArgumentCaptor<Supplier<Transport>> transportSupplierCaptor;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    loadBalancer = RoundRobinLoadBalancerFactory.getInstance().newLoadBalancer(
+        "fakeservice", mockTransportManager);
+    addressGroupList = new ArrayList<EquivalentAddressGroup>();
+    servers = new ArrayList<List<ResolvedServerInfo>>();
+    for (int i = 0; i < 3; i++) {
+      servers.add(new ArrayList<ResolvedServerInfo>());
+      ArrayList<SocketAddress> addresses = new ArrayList<SocketAddress>();
+      for (int j = 0; j < 3; j++) {
+        SocketAddress addr = new FakeSocketAddress("servergroup" + i + "server" + j);
+        servers.get(i).add(new ResolvedServerInfo(addr, Attributes.EMPTY));
+        addresses.add(addr);
+      }
+      addressGroupList.add(new EquivalentAddressGroup(addresses));
+    }
+    when(mockTransportManager.getTransport(eq(addressGroupList.get(0))))
+        .thenReturn(mockTransport0);
+    when(mockTransportManager.getTransport(eq(addressGroupList.get(1))))
+        .thenReturn(mockTransport1);
+    when(mockTransportManager.getTransport(eq(addressGroupList.get(2))))
+        .thenReturn(mockTransport2);
+    when(mockTransportManager.createInterimTransport()).thenReturn(mockInterimTransport);
+    when(mockInterimTransport.transport()).thenReturn(mockInterimTransportAsTransport);
+  }
+
+  @Test
+  public void pickBeforeResolved() throws Exception {
+    Transport t1 = loadBalancer.pickTransport(null);
+    Transport t2 = loadBalancer.pickTransport(null);
+    assertSame(mockInterimTransportAsTransport, t1);
+    assertSame(mockInterimTransportAsTransport, t2);
+    verify(mockTransportManager).createInterimTransport();
+    verify(mockTransportManager, never()).getTransport(any(EquivalentAddressGroup.class));
+    verify(mockInterimTransport, times(2)).transport();
+
+    loadBalancer.handleResolvedAddresses(servers, Attributes.EMPTY);
+    verify(mockInterimTransport).closeWithRealTransports(transportSupplierCaptor.capture());
+    assertSame(mockTransport0, transportSupplierCaptor.getValue().get());
+    assertSame(mockTransport1, transportSupplierCaptor.getValue().get());
+    verify(mockTransportManager, times(1)).getTransport(eq(addressGroupList.get(0)));
+    verify(mockTransportManager, times(1)).getTransport(eq(addressGroupList.get(1)));
+    verify(mockTransportManager, times(1)).createFailingTransport(any(Status.class));
```

Comment:
You may want to use [inOrder](http://site.mockito.org/mockito/docs/1.10.19/org/mockito/Mockito.html#in_order_verification)